cf/regrid/regrid.py

```diff
@@ -0,0 +1,1895 @@
+
+
+def update_non_coordinates(
+    src, dst, regrid_operator, src_grid=None, dst_grid=None
+):
+    """Update the coordinate references of the regridded field.
+
+    :Parameters:
+
+        src: `Field`
+            The regridded field. Updated in-place.
+
+        dst: `Field` or `Domain`
+            The field or domain containing the destination grid.
+
+        regrid_operator: `RegridOperator`
+            The regrid operator.
+
+        src_grid: `Grid`
+            The definition of the source grid.
+
+        dst_grid: `Grid`
+            The definition of the destination grid.
+
+    :Returns:
+
+        `None`
+
+    """
+    src_axis_keys = src_grid.axis_keys
+    dst_axis_keys = dst_grid.axis_keys
+
+    # Initialise cached value for domain_axes
+    domain_axes = None
+
+    data_axes = src.constructs.data_axes()
+
+    # ----------------------------------------------------------------
+    # Delete source grid coordinate references whose `coordinates`
+    # span any of the regridding axes. Also delete the corresponding
+    # domain ancillaries.
+    # ----------------------------------------------------------------
+    for ref_key, ref in src.coordinate_references(todict=True).items():
+        ref_axes = []
+        for c_key in ref.coordinates():
+            ref_axes.extend(data_axes[c_key])
+
+        if set(ref_axes).intersection(src_axis_keys):
+            src.del_coordinate_reference(ref_key)
+
+    # ----------------------------------------------------------------
+    # Delete source grid cell measures and field ancillaries that span
+    # any of the regridding axes
+    # ----------------------------------------------------------------
+    for key in src.constructs(
+        filter_by_type=("cell_measure", "field_ancillary"), todict=True
+    ):
+        if set(data_axes[key]).intersection(src_axis_keys):
+            src.del_construct(key)
+
+    # ----------------------------------------------------------------
+    # Regrid any remaining source domain ancillaries that span all of
+    # the regridding axes
+    # ----------------------------------------------------------------
+    for da_key in src.domain_ancillaries(todict=True):
+        da_axes = data_axes[da_key]
+
+        # Ignore any remaining source domain ancillary that spans none
+        # of the regridding axes
+        if not set(da_axes).intersection(src_axis_keys):
+            continue
+
+        # Delete any any remaining source domain ancillary that spans
+        # some but not all of the regridding axes
+        if not set(da_axes).issuperset(src_axis_keys):
+            src.del_construct(da_key)
+            continue
+
+        # Convert the domain ancillary to a field, without any
+        # non-coordinate metadata (to prevent them being unnecessarily
+        # processed during the regridding of the domain ancillary
+        # field, and especially to avoid potential
+        # regridding-of-domain-ancillaries infinite recursion).
```

Comment:
> Since we've updated the `TODODASKVER` now on the `lama-to-dask` branch,
you might want to blitz those introduced here before merging.

That sounds nasty :grimacing: